File: elixir/anagram/anagram.exs
defmodule Anagram do
  @doc """
  Returns all candidates that are anagrams of, but not equal to, 'base'.
  """
  @spec match(String.t, [String.t]) :: [String.t]
  def match(base, candidates) do
    Enum.filter(candidates, &(is_anagram?(&1, base)))
  end

  defp is_anagram?(candidate, base) do
    letter_counts(candidate) == letter_counts(base) &&
      String.downcase(candidate) != String.downcase(base)
  end

  def letter_counts(word) do
    word |> String.downcase |> String.codepoints |> Enum.sort
  end
end

File: ruby/anagram/anagram.rb
class Anagram
  def initialize(word)
    @word       = word.downcase
    @word_chars = @word.chars.sort
  end

  def match(candidates)
    candidates.select do |i|
      i = i.downcase
      i.chars.sort == @word_chars && i != @word
    end
  end
end
